serverless/event_workflows/serverless-subs.adoc

```diff
@@ -0,0 +1,30 @@
+include::modules/serverless-document-attributes.adoc[]
+[id="serverless-subs"]
+= Using subscriptions
+include::modules/common-attributes.adoc[]
+:context: serverless-subs
+
+toc::[]
+
+After events have been sent to a channel from an event source or producer, these events can be sent to multiple Knative services, or other sinks, by using a subscription.
+
+image::serverless-event-channel-workflow.png[Channel workflow overview]
+
+If a subscriber rejects an event, there are no re-delivery attempts by default. Developers can configure xref:../../serverless/event_workflows/serverless-event-delivery.adoc#serverless-event-delivery[re-delivery attempts] by modifying the `delivery` spec in a `Subscription` object.
+
+[id="serverless-subs-creating-subs"]
+== Creating subscriptions
+
+Developers can create subscriptions that allow event sinks to subscribe to channels and receive events.
+
+include::modules/serverless-creating-subscriptions-odc.adoc[leveloffset=+2]
+include::modules/serverless-creating-subscriptions-kn.adoc[leveloffset=+2]
+include::modules/serverless-creating-subscriptions-yaml.adoc[leveloffset=+2]
+// config event delivery
+include::modules/serverless-subscription-event-delivery-config.adoc[leveloffset=+1]
```

Comment:
Why adding this section as top-level when it only applies to "Creating subscriptions" ? The top-level as shown here is reserved so far fro CRUD operations that are all on the same level. Addin a minor and specialized topic on the same level is confusing:

![image](https://user-images.githubusercontent.com/99080/112125295-8d462480-8bc3-11eb-9d76-ad216c22e874.png)

Side Note: Updating subscriptions is missing in this list, too.

Reply:
The idea here was that event delivery can be configured at any time, it's not strictly only part of the creation of a subscription, especially when the only current method of configuring this is YAML, so if a user creates a subscription via ODC for example, they won't afaik see any options for this, and would need to first create the subscription in the Topology view, and then go and edit the YAML for it in ODC, rather than being able to add that config during creation. Is this not correct?

Re updating, this wasn't requested in the Jira or mentioned in the comments for it, nor is it in the knative.dev docs, but the options look the same as those for create, so I'll add it for here and upstream.

Reply:
I see the point, but then I still would move that item to the end of the list as "CRUD" (Create-Retrieve-Update-Delete) is a typical way how people look at it, so those tasks should go together.

----
Unrelated comment, I will add a separate JIRA for this:
The description "Configuring event delivery failure parameters using subscriptions" is so long, because of the context required so that it can be found ? I mean "using subscriptions" is quite redundant here (it's in the headline already), but probably needed for technical reasons (the TOC entries can't be different than the headlines, right ?). Same goes for "using the Knative CLI". This could be just part of a "Describing subscriptions" section (we could also shorten the "Creating subscriptions by using ..." to "Creating by using ...." or just use `.headings` so that it doesn't show up in the TOC.

Also not sure why there is a https://docs.openshift.com/container-platform/4.7/serverless/admin_guide/serverless-cluster-admin-eventing.html#serverless-creating-subscription-admin-web-console_serverless-cluster-admin-eventing which also takes about creating subscription but from the Admin perspective? Shouldn't it be on this page, too ? (if at all as I consider creating services, subscriptions etc. not to be an Admin concern at all)

Reply:
> I see the point, but then I still would move that item to the end of the list as "CRUD" (Create-Retrieve-Update-Delete) is a typical way how people look at it, so those tasks should go together.

Doesn't that make _less_ sense in terms of UX though? Wouldn't event delivery be something they're more likely to want to do immediately after creating than after updating or deleting? I don't think it makes sense to move this tbh.

> The description "Configuring event delivery failure parameters using subscriptions" is so long, because of the context required so that it can be found ? I mean "using subscriptions" is quite redundant here (it's in the headline already), but probably needed for technical reasons (the TOC entries can't be different than the headlines, right ?).

This has nothing to do with TOCs. This is because the modules need to have properly descriptive titles for content re-use. The whole point of mod docs is that the content can be included in multiple places, so we should never assume for modules that context can be provided by the assembly title. For example, this module is also included in the event delivery section: https://deploy-preview-30282--osdocs.netlify.app/openshift-enterprise/latest/serverless/event_workflows/serverless-event-delivery.html and needs to be specific because there will eventually be an option to configure this using triggers for brokers too etc, or maybe even via configmaps etc for global config. We need to be specific about how it's configured so users can find it easily from the heading.

>   Same goes for "using the Knative CLI". This could be just part of a "Describing subscriptions" section (we could also shorten the "Creating subscriptions by using ..." to "Creating by using ...." or just use .headings so that it doesn't show up in the TOC.

Again, there are multiple ways to do this, so we need to state in the title this is the Knative CLI method. Re using .headings, that's not how mod docs work. There should only ever be one procedure per module. You can check out the guidelines for more info on that.

> Also not sure why there is a https://docs.openshift.com/container-platform/4.7/serverless/admin_guide/serverless-cluster-admin-eventing.html#serverless-creating-subscription-admin-web-console_serverless-cluster-admin-eventing which also takes about creating subscription but from the Admin perspective? Shouldn't it be on this page, too ? (if at all as I consider creating services, subscriptions etc. not to be an Admin concern at all)

No, there is a specific Eventing tab in the nav of the Admin console that doesn't show in the Developer perspective of ODC. It was requested that we doc this. If you go and look at the Admin tab after installing serverless on a cluster you'll see it.

Reply:
@rhuss please direct any other additional comments that are unrelated to the actual technical content in this PR into Jiras or raise them in Slack or on a call.

Reply:
done